### PR TITLE
Feat/secrets management edge functions

### PR DESCRIPTION
**Pull request by joshenlim**

### CODE REVIEW

**Comment by inian** [comment]
We shouldn't show the delete button for [reserved secrets](https://github.com/supabase/infrastructure/blob/cc4375015c91d14ee263c0447957c173e2d1d7e0/api/src/routes/v1/projects/ref/secrets/secrets.controller.ts#L126)

**Review by alaister — CHANGES_REQUESTED** [review]
Looking great! Found a couple small things:

File: studio/components/interfaces/Functions/EdgeFunctionSecrets/EdgeFunctionSecret.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                <IconEyeOff size={16} strokeWidth={1.5} />
+              ) : (
+                <IconEye size={16} strokeWidth={1.5} />
+              )
+            }
+            className="px-1"
+            onClick={() => setShow(!show)}
+          />
+          {show ? (
+            <Input copy value={secret.value} size="small" className="font-mono w-full" />
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply studio/components/interfaces/Functions/EdgeFunctionSecrets/EdgeFunctionSecret.tsx:34]: ahh yes - inian left a comment on this too that this is the "digest" i'll need to rename the headers and won't need to hide the values
behaviour should match CLI

File: studio/components/interfaces/Functions/EdgeFunctionSecrets/EdgeFunctionSecrets.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  const [selectedSecret, setSelectedSecret] = useState<ProjectSecret>()
+
+  const {
+    data: secrets,
+    error,
+    isLoading,
+    isSuccess,
+    isError,
+  } = useSecretsQuery({
+    projectRef: projectRef ?? '',
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply studio/components/interfaces/Functions/EdgeFunctionSecrets/EdgeFunctionSecrets.tsx:28]: ahh yeah this shouldn't be on local/self-hosted

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**